New module: HiGal catalog + image cutout service

keflavich opened the pull request:
This is a new tool to query the HiGal image cutout and catalog service.  The catalog can be accessed from Vizier and therefore is not necessary, but the image cutout service is unique.

The service is hosted at https://tools.ssdc.asi.it/HiGALSearch.jsp.  I have never been able to find that URL by googling it.

I still need to make some local tests, but remote tests exist.

keflavich committed:
# This is a combination of 13 commits.
# This is the 1st commit message:

add HiGal module

more cleanup

add changelog

it is possible for a cutout to be missing; we need to allow for that.
Also, the hack I had before isn't needed when you appropriately query
for multiple wavelengths

remove some comments, still not done with the PR

minor correction

added basic docs

pep8 fixes.  The bracket closures DO match the indentation, though.
pep8 lies sometimes.

add local tests and cleanup some testing

more interactive tests revealed that my hacks didn't really work, this
commit has some fixes

move everything into astroquery.herschel.higal

update relative paths

whitespace

# This is the commit message #2:

update tests

# This is the commit message #3:

the higal server genuinely isn't showing anything when I search online now

# This is the commit message #4:

add HiGal module

# This is the commit message #5:

more cleanup

# This is the commit message #6:

add changelog

# This is the commit message #7:

it is possible for a cutout to be missing; we need to allow for that.
Also, the hack I had before isn't needed when you appropriately query
for multiple wavelengths

# This is the commit message #8:

remove some comments, still not done with the PR

# This is the commit message #9:

minor correction

# This is the commit message #10:

added basic docs

# This is the commit message #11:

pep8 fixes.  The bracket closures DO match the indentation, though.
pep8 lies sometimes.

# This is the commit message #12:

add local tests and cleanup some testing

# This is the commit message #13:

more interactive tests revealed that my hacks didn't really work, this
commit has some fixes

keflavich commented:
TODO:

* [ ] documentation
* [ ] local tests

bsipocz commented:
what other stuff lives at ssdc? Do we foresee that any of those will need a module in the future? If yes, in that case it may make sense to restructure from the beginning and call it ssdc and have higal under it.

keflavich commented:
Good questions.  I have no clue.  The sub-interface for HiGal appears to be at least somewhat distinct from the general SSDC querier.  We need one of them onboard....

keflavich commented:
An interesting deficiency I discovered working on this is that `astropy.utils.data.get_readable_fileobj` does not work for files that require a cookie.  That's an oversight we could possibly handle in the distant future.

bsipocz commented:
what is the status of the other Herschel surveys then? maybe a herschel module then? Though that would be far from any homogeneity.

keflavich commented:
unfortunately, it looks like Herschel surveys are hosted randomly.  Many are at IPAC, others are not.

pep8speaks commented:
Hello @keflavich! Thanks for updating the PR.



 - In the file [`astroquery/higal/core.py`](https://github.com/astropy/astroquery/blob/0e397164d8116625303e3a359857ba2dee915982/astroquery/higal/core.py), following are the PEP8 issues :

> [Line 40:5](https://github.com/astropy/astroquery/blob/0e397164d8116625303e3a359857ba2dee915982/astroquery/higal/core.py#L40): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 41:5](https://github.com/astropy/astroquery/blob/0e397164d8116625303e3a359857ba2dee915982/astroquery/higal/core.py#L41): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 47:22](https://github.com/astropy/astroquery/blob/0e397164d8116625303e3a359857ba2dee915982/astroquery/higal/core.py#L47): [E124](https://duckduckgo.com/?q=pep8%20E124) closing bracket does not match visual indentation
> [Line 53:25](https://github.com/astropy/astroquery/blob/0e397164d8116625303e3a359857ba2dee915982/astroquery/higal/core.py#L53): [E124](https://duckduckgo.com/?q=pep8%20E124) closing bracket does not match visual indentation
> [Line 88:48](https://github.com/astropy/astroquery/blob/0e397164d8116625303e3a359857ba2dee915982/astroquery/higal/core.py#L88): [E124](https://duckduckgo.com/?q=pep8%20E124) closing bracket does not match visual indentation
> [Line 125:9](https://github.com/astropy/astroquery/blob/0e397164d8116625303e3a359857ba2dee915982/astroquery/higal/core.py#L125): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (2)
> [Line 130:26](https://github.com/astropy/astroquery/blob/0e397164d8116625303e3a359857ba2dee915982/astroquery/higal/core.py#L130): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 132:41](https://github.com/astropy/astroquery/blob/0e397164d8116625303e3a359857ba2dee915982/astroquery/higal/core.py#L132): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 132:42](https://github.com/astropy/astroquery/blob/0e397164d8116625303e3a359857ba2dee915982/astroquery/higal/core.py#L132): [E262](https://duckduckgo.com/?q=pep8%20E262) inline comment should start with '# '
> [Line 145:39](https://github.com/astropy/astroquery/blob/0e397164d8116625303e3a359857ba2dee915982/astroquery/higal/core.py#L145): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 234:48](https://github.com/astropy/astroquery/blob/0e397164d8116625303e3a359857ba2dee915982/astroquery/higal/core.py#L234): [E124](https://duckduckgo.com/?q=pep8%20E124) closing bracket does not match visual indentation
> [Line 251:22](https://github.com/astropy/astroquery/blob/0e397164d8116625303e3a359857ba2dee915982/astroquery/higal/core.py#L251): [E124](https://duckduckgo.com/?q=pep8%20E124) closing bracket does not match visual indentation





 - In the file [`astroquery/higal/tests/test_higal_remote.py`](https://github.com/astropy/astroquery/blob/0e397164d8116625303e3a359857ba2dee915982/astroquery/higal/tests/test_higal_remote.py), following are the PEP8 issues :

> [Line 11:1](https://github.com/astropy/astroquery/blob/0e397164d8116625303e3a359857ba2dee915982/astroquery/higal/tests/test_higal_remote.py#L11): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1

bsipocz commented:
> An interesting deficiency I discovered working on this is that `astropy.utils.data.get_readable_fileobj` does not work for files that require a cookie. That's an oversight we could possibly handle in the distant future.

Apparently I haven't noticed this comment before. Do you suggest that we handle that here, or upstream? If the latter can you open an issue for astropy?

keflavich commented:
I guess upstream?  But how often is this actually needed?  I'm not really sure how we'd handle it upstream anyway; `requests` makes it so much easier I might just say we should require it for cookie-related downloads.

So... not upstream, then!

bsipocz commented:
preferring to use`requests` sounds actually super sensible.

kakirastern commented:
Can I help with this PR to help bring it to fruition (as suggested by @keflavich earlier)? If I am to finish up this PR I might need to open an other PR to "supplant" this...

keflavich commented:
Yes, go ahead and work on this one if you like.  I don't remember where I left it off, but maybe all we need to do is get tests passing?

You can submit a new PR based on this one.  Normally I'd prefer to just give write permissions, but I don't know how to do that.

kakirastern commented:
Yeah, I would prefer to get write permissions too. That would be easier for me as well. Would this help?: [https://help.github.com/en/articles/managing-an-individuals-access-to-an-organization-repository](https://help.github.com/en/articles/managing-an-individuals-access-to-an-organization-repository)

keflavich commented:
Right, I forgot I can just give you write permissions to _my_ fork - we don't (and can't) give out permission to write to `astropy/astroquery`.  Anyway, you should now have permission to write to this branch.

kakirastern commented:
No problem, I can add new changes to this PR via your fork... Will follow up on this very soon.

bsipocz commented on docs/higal/higal.rst:
```diff
@@ -0,0 +1,51 @@
+
+An example catalog query:
+
+.. python::
```
either remove this (should work without) or add `.. code-block:: python`

kakirastern replied:
Changed to `.. code-block:: python`

kakirastern replied:
Added blank line after this line

bsipocz commented on docs/higal/higal.rst:
```diff
@@ -0,0 +1,51 @@
+
+And an example image query:
+
+.. python::
```
same here

kakirastern replied:
Changed to `.. code-block:: python`

kakirastern replied:
Added blank line after this line

bsipocz commented on docs/higal/higal.rst:
```diff
@@ -0,0 +1,51 @@
+Reference/API
+=============
+
+.. automodapi:: astroquery.higal
```
also link this file in the main index docs.

kakirastern replied:
Added `higal/higal.rst` to `index.rst`

bsipocz reviewed (changes requested):
Some minor things that should fix the failing docs build

bsipocz commented:
Organizing:should we start it here before it goes in?


@keflavich - would you be against an `astroquery.herschel.higal` rather than having it in the top namespace?

keflavich commented:
No, `.herschel.higal` is fine.  It might also end up co-listed under `esac.higal` and `cutout_services.higal` eventually?

bsipocz commented:
`esa.higal` and `cutout_services.higal` sounds very good approach to me. The only trick is that we need to make sure users know it's all the same whichever namespace they end up using.

kakirastern commented:
All checks have passed, ready for the next stage...

kakirastern commented:
Should I change the organization to `astroquery.herschel.higal` next? I can change both the names for the docs and the code...

bsipocz reviewed (changes requested):
This needs to be put in the new namespacing scheme right away, so we won't need to move the module around.

keflavich commented:
Just noting that.... we still have to finish the namespace refactor. 😱

bsipocz commented:
@keflavich - Is this still something on the table? I also wonder whether it should be part of more ssdc functionality (and namespace) rather than to exists on its own?

bsipocz commented:
I pulled up this PR as I'm at a talk about SSDC as is, and it seems that there are more services and tools they provide, many with a VO. But I haven't gone into the weeds of how much of that we could/want/should support.

keflavich commented:
Oh wow, great!  If there are other SSDC products, then, I'm super happy to support them!  I hope some of them have nicer APIs - VO would be great.

reminder to me: https://www.ssdc.asi.it/ is the SSDC homepage.

keflavich committed:
add HiGal module

more cleanup

add changelog

it is possible for a cutout to be missing; we need to allow for that.
Also, the hack I had before isn't needed when you appropriately query
for multiple wavelengths

remove some comments, still not done with the PR

minor correction

added basic docs

pep8 fixes.  The bracket closures DO match the indentation, though.
pep8 lies sometimes.

add local tests and cleanup some testing

more interactive tests revealed that my hacks didn't really work, this
commit has some fixes

move everything into astroquery.herschel.higal

update relative paths

whitespace

update tests

the higal server genuinely isn't showing anything when I search online now

add HiGal module

more cleanup

add changelog

it is possible for a cutout to be missing; we need to allow for that.
Also, the hack I had before isn't needed when you appropriately query
for multiple wavelengths

remove some comments, still not done with the PR

minor correction

added basic docs

pep8 fixes.  The bracket closures DO match the indentation, though.
pep8 lies sometimes.

add local tests and cleanup some testing

more interactive tests revealed that my hacks didn't really work, this
commit has some fixes

move everything into astroquery.herschel.higal

keflavich committed:
add HiGal module

more cleanup

add changelog

it is possible for a cutout to be missing; we need to allow for that.
Also, the hack I had before isn't needed when you appropriately query
for multiple wavelengths

remove some comments, still not done with the PR

minor correction

added basic docs

pep8 fixes.  The bracket closures DO match the indentation, though.
pep8 lies sometimes.

add local tests and cleanup some testing

more interactive tests revealed that my hacks didn't really work, this
commit has some fixes

move everything into astroquery.herschel.higal

update relative paths

whitespace

update tests

the higal server genuinely isn't showing anything when I search online now

add HiGal module

more cleanup

add changelog

it is possible for a cutout to be missing; we need to allow for that.
Also, the hack I had before isn't needed when you appropriately query
for multiple wavelengths

remove some comments, still not done with the PR

minor correction

added basic docs

pep8 fixes.  The bracket closures DO match the indentation, though.
pep8 lies sometimes.

add local tests and cleanup some testing

more interactive tests revealed that my hacks didn't really work, this
commit has some fixes

move everything into astroquery.herschel.higal

update relative paths

whitespace

keflavich commented:
I've marked this as a 'draft pr' to prevent merging.

The HiGal service doesn't work for me right now via browser.
